src/main/java/org/opentripplanner/middleware/trip_monitor/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,176 @@
+        // TODO
+    }
+
+    private void checkTripForDelays(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+
+    private void sendNotification() {
+        if (notifications.size() == 0) {
+            // FIXME: Change log level
+            LOG.info("No notifications queued up. Skipping notify.");
+            return;
+        }
+        OtpUser otpUser = Persistence.otpUsers.getById(trip.userId);
+        if (otpUser == null) {
+            LOG.error("Cannot find user for id {}", trip.userId);
+            // TODO: Bugsnag / delete monitored trip?
+            return;
+        }
+        String name = trip.tripName != null ? trip.tripName : "Trip for " + otpUser.email;
+        String subject = name + " Notification";
+        StringBuilder body = new StringBuilder();
+        for (TripMonitorNotification notification : notifications) {
+            body.append(notification.body);
+        }
+        // FIXME: Change log level
+        LOG.info("Sending notification '{}' to user {}", subject, trip.userId);
+        // FIXME: This needs to be an enum.
+        switch (otpUser.notificationChannel.toLowerCase()) {
+            case "sms":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                break;
+            case "email":
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void checkTripForNewAlerts(MonitoredTrip trip, Itinerary itinerary) {
+        if (!trip.notifyOnAlert) {
+            LOG.debug("Notify on alert is disabled for trip {}. Skipping check.", trip.id);
+            return;
+        }
+        // FIXME: Need to somehow determine how to get last itinerary from responses in journey state.
+        Itinerary lastItinerary = trip.lastItinerary();
+        List<LocalizedAlert> newAlerts = itinerary.getAlerts();
+        List<LocalizedAlert> previousAlerts = lastItinerary.getAlerts();
+        // If journey state is already tracking alerts from previous checks, see if they have changed.
+        if (previousAlerts.size() > 0) {
+            // If there are new alerts, notify user.
+            List<LocalizedAlert> unseenAlerts = new ArrayList<>();
+            for (LocalizedAlert newAlert : newAlerts) {
+                boolean alertIsNew = true;
+                for (LocalizedAlert previousAlert : previousAlerts) {
+                    if (newAlert.equals(previousAlert)) {
+                        alertIsNew = false;
+                        break;
+                    }
+                }
+                if (alertIsNew) unseenAlerts.add(newAlert);
+            }
```

Comment:
Seems like you could make a Set of LocalizedAlerts and then just call `removeAll` to have the same effect as this code.